### PR TITLE
BC-2398 - long running course copies

### PR DESCRIPTION
**Pull request by hoeppner-dataport**
# Short Description
When course copies take longer than the defined server request timeout, users were only informed, that the copy continues in the background.
Now they are also redirected to the course overview page and the course being copied is displayed as such and not yet accessible. When the user stays on this screen the courses-data is pulled periodically - until the process was finished.
User is informed by a success-message then.

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/BC-2398
https://github.com/hpi-schul-cloud/schulcloud-server/pull/3865

## Screenshots of UI changes
BC-2398-long-running-course-copies

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/pages/RoomOverview.page.vue [hunk 1]
```diff
@@ -366,6 +373,22 @@ export default {
 			this.$router.replace({ path: "/rooms-overview" });
 			roomsModule.fetch();
 		},
+		reloadCourses(count = 0, started) {
```
Comment (OliverHappe) [inline src/pages/RoomOverview.page.vue:376]: I think `reloadCourses` does not describe well enough that this function will keep on going and is not just a singular call to the API. Maybe one of these names is better? 

`reloadCoursesRecursively`

`initCoursePolling`

File: src/composables/copy.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { CopyApiResponseStatusEnum } from "@/serverApi/v3";
 import CopyModule, { CopyParams, CopyParamsTypeEnum } from "@/store/copy";
 import NotifierModule from "@/store/notifier";
-import { inject, InjectionKey, Ref } from "@vue/composition-api";
+import { inject, InjectionKey, Ref, reactive } from "@vue/composition-api";
 import VueI18n from "vue-i18n";
 
 export const USE_COPY: InjectionKey<typeof useCopy> = Symbol();
```
Comment (OliverHappe) [inline src/composables/copy.ts:7]: unused line

File: src/pages/RoomOverview.page.vue (hunk 1 above)
Comment (hoeppner-dataport) [reply src/pages/RoomOverview.page.vue]: fixed

File: src/composables/copy.ts (hunk 2 above)
Comment (hoeppner-dataport) [reply src/composables/copy.ts]: fixed

**Review by OliverHappe — APPROVED** [review]

**Review by uidp — DISMISSED** [review]

**Review by OliverHappe — DISMISSED** [review]

**Pull request merged**